tests/slsa_analyzer/checks/test_registry.py

```diff
@@ -200,3 +254,112 @@ def test_validate_check_id_format(self) -> None:
+        ),
+        (["mcn_c_1"], ["*"], ["mcn_d_1", "mcn_e_1", "mcn_g_1", "mcn_h_1", "mcn_i_1"]),
+        ([], ["mcn_c_1"], ["mcn_c_1", "mcn_d_1", "mcn_e_1"]),
+        (["mcn_d_1"], ["mcn_c_1"], ["mcn_e_1"]),
+        (["mcn_*"], ["*"], []),
+    ],
+)
+def test_get_final_checks(
+    check_registry: Registry, ex_pats: list[str], in_pats: list[str], final_checks: list[str]
+) -> None:
+    """This method tests the get_final_checks method."""
+    assert check_registry.get_final_checks(ex_pats=ex_pats, in_pats=in_pats).sort() == final_checks.sort()
+
+
+@pytest.mark.parametrize(
+    ("parent_id", "children"),
+    [
+        ("mcn_c_1", ["mcn_c_1", "mcn_f_1", "mcn_b_1", "mcn_a_1"]),
+        # ("mcn_a_1", ["mcn_a_1"]),
+    ],
+)
+def test_get_transitive_children(check_registry: Registry, parent_id: str, children: list[str]) -> None:
+    """This method test the get_transitive_children method."""
+    result = sorted(check_registry.get_transitive_children(parent_id))
+    expect = sorted(children)
+    assert result == expect
+
+
+@pytest.mark.parametrize(
+    ("child_id", "parents"),
+    [
+        ("mcn_c_1", ["mcn_c_1", "mcn_d_1", "mcn_e_1"]),
+        ("mcn_a_1", ["mcn_a_1", "mcn_b_1", "mcn_c_1", "mcn_d_1", "mcn_e_1"]),
+        ("mcn_e_1", ["mcn_e_1"]),
+    ],
+)
+def test_get_transitive_parents(check_registry: Registry, child_id: str, parents: list[str]) -> None:
+    """This method test the get_transitive_parents method."""
+    result = sorted(check_registry.get_transitive_parents(child_id))
+    expect = sorted(parents)
+    assert result == expect
+
+
+def test_get_children_parents_special_cases(check_registry: Registry) -> None:
+    """This method test the special cases for getting transitive child or parent checks."""
+    with pytest.raises(CheckRegistryError):
+        check_registry.get_transitive_parents("not_exist")
+
+    with pytest.raises(CheckRegistryError):
+        check_registry.get_transitive_children("not_exist")
+
+
+@pytest.mark.parametrize(
+    ("parent_id", "has_error"),
+    [("mcn_a_1", True), ("mcn_b_1", True), ("mcn_d_1", False)],
+)
+def test_get_transitive_children_circular(circular_registry: Registry, parent_id: str, has_error: bool) -> None:
+    """This method test get_transitive_children method with circular detection"""
+    if has_error:
+        with pytest.raises(CheckRegistryError):
+            circular_registry.get_transitive_children(parent_id)
+    else:
+        assert circular_registry.get_transitive_children(parent_id)
+
+
+@pytest.mark.parametrize(
+    ("child_id", "has_error"),
+    [("mcn_a_1", True), ("mcn_b_1", True), ("mcn_d_1", False)],
+)
+def test_get_transitive_parent_circular(circular_registry: Registry, child_id: str, has_error: bool) -> None:
+    """This method test get_transitive_parent method with circular detection"""
+    if has_error:
+        with pytest.raises(CheckRegistryError):
```

Comment:
Detecting `CheckRegistryError` here is not correct as it's the parent of `CheckCircularDependencyError` and `CheckNotExistError`. I will update the tests which detect `CheckRegistryError` to detect the specific exception they are testing for.

Reply:
See https://github.com/oracle/macaron/pull/254/commits/7dde1a163871bb9d6029fe2635dd957b5c252721